### ISSUE TITLE
DCS crashes on start

### ISSUE THREAD

**Issue by daniellovell** [mbucchia/Pimax-OpenXR#21]
Here is the error:
![image](https://user-images.githubusercontent.com/6250953/184472922-5a4dae27-0fd5-480d-be51-eefe6c00b1e0.png)

Here is the pimax-openxr.log: 
```2022-08-12 23:57:41 -0700: PimaxXR - v0.2.4
2022-08-12 23:57:41 -0700: PVR: 1.23.1
2022-08-12 23:57:41 -0700: Application: OpenComposite_DCS; Engine: 
2022-08-12 23:57:42 -0700: PVR: 1.23.1
2022-08-12 23:57:42 -0700: Application: OpenComposite_DCS; Engine: 
2022-08-12 23:57:42 -0700: Requested extension: XR_KHR_D3D11_enable
2022-08-12 23:57:42 -0700: Requested extension: XR_KHR_D3D12_enable
2022-08-12 23:57:42 -0700: Requested extension: XR_KHR_vulkan_enable
2022-08-12 23:57:42 -0700: Requested extension: XR_KHR_visibility_mask
2022-08-12 23:57:42 -0700: Requested extension: XR_KHR_win32_convert_performance_counter_time
2022-08-12 23:57:44 -0700: Using Vulkan on adapter: NVIDIA GeForce RTX 3080
2022-08-12 23:57:44 -0700: Parallel projection is enabled
2022-08-12 23:57:44 -0700: xrLocateViews: XrResult failure [XR_ERROR_HANDLE_INVALID]
    Origin: xrLocateSpace(m_viewSpace, viewLocateInfo->space, viewLocateInfo->displayTime, &location)
    Source: D:\a\Pimax-OpenXR\Pimax-OpenXR\pimax-openxr\space.cpp:281
2022-08-12 23:57:44 -0700: xrLocateViews failed with XR_ERROR_RUNTIME_FAILURE
2022-08-12 23:57:44 -0700: xrLocateViews: XrResult failure [XR_ERROR_HANDLE_INVALID]
    Origin: xrLocateSpace(m_viewSpace, viewLocateInfo->space, viewLocateInfo->displayTime, &location)
    Source: D:\a\Pimax-OpenXR\Pimax-OpenXR\pimax-openxr\space.cpp:281
2022-08-12 23:57:44 -0700: xrLocateViews failed with XR_ERROR_RUNTIME_FAILURE
2022-08-12 23:57:48 -0700: xrLocateViews: XrResult failure [XR_ERROR_HANDLE_INVALID]
    Origin: xrLocateSpace(m_viewSpace, viewLocateInfo->space, viewLocateInfo->displayTime, &location)
    Source: D:\a\Pimax-OpenXR\Pimax-OpenXR\pimax-openxr\space.cpp:281
2022-08-12 23:57:48 -0700: xrLocateViews failed with XR_ERROR_RUNTIME_FAILURE
2022-08-12 23:57:48 -0700: xrLocateViews: XrResult failure [XR_ERROR_HANDLE_INVALID]
    Origin: xrLocateSpace(m_viewSpace, viewLocateInfo->space, viewLocateInfo->displayTime, &location)
    Source: D:\a\Pimax-OpenXR\Pimax-OpenXR\pimax-openxr\space.cpp:281
2022-08-12 23:57:48 -0700: xrLocateViews failed with XR_ERROR_RUNTIME_FAILURE
2022-08-12 23:57:48 -0700: xrLocateViews: XrResult failure [XR_ERROR_HANDLE_INVALID]
    Origin: xrLocateSpace(m_viewSpace, viewLocateInfo->space, viewLocateInfo->displayTime, &location)
    Source: D:\a\Pimax-OpenXR\Pimax-OpenXR\pimax-openxr\space.cpp:281
2022-08-12 23:57:48 -0700: xrLocateViews failed with XR_ERROR_RUNTIME_FAILURE
```
I believe I have followed all the installation instructions correctly. I have retried with a complete uninstallation and reinstallation of OpenComposite, Pimax-OpenXR, and DCS to confirm that I followed all installation steps

**Comment by daniellovell** [comment]
And here is the error I receive if I try to capture a trace:
![companion_48aFQnaIjN](https://user-images.githubusercontent.com/6250953/184473230-f50ce9dc-5ed8-4c5e-8d6d-43140fc9d1c1.png)

**Comment by daniellovell** [comment]
Ah, I just took a quick look at the code in space.cpp:281. I think I have a hunch what's going on: I'm not using base stations with my Pimax headset, just the integrated 9-axis IMU.

**Comment by daniellovell** [comment]
I tried to do a little bit of debugging, here is what I found - hope it helps:
1. The capture trace bug seems to go away when I run as administrator, but the final trace gets corrupted anyway:
![image](https://user-images.githubusercontent.com/6250953/184480662-49569b5e-4120-46d8-973c-a3c3924849b4.png)
I'll email you the ETL file if you want to see if it opens on your side (just wont upload here since personal info, etc)
2. So I put some fprintfs in to take a peek at what was going on, here is the output:
[pimax-openxr-DUMP.txt](https://github.com/mbucchia/Pimax-OpenXR/files/9331293/pimax-openxr-DUMP.txt)
OpenComposite seems to be be alternating between well-formed calls to xrLocateView and calls with a null `ViewLocateInfo->space` specified. I took a look at OpenComposite's source and cannot immediately see how that is possible. I am running the latest OpenComposite.
3. Incidentally, I spotted some typos in space.cpp:
https://github.com/mbucchia/Pimax-OpenXR/blob/75c575d110ddbd4529c0b09fa9ef51ae01e4baec/pimax-openxr/space.cpp#L198
https://github.com/mbucchia/Pimax-OpenXR/blob/75c575d110ddbd4529c0b09fa9ef51ae01e4baec/pimax-openxr/space.cpp#L206
Probably meant `flags2` for that second `IsPoseTracked` check. Not worth a PR for something so small, but wanted to point it out.
Cheers

**Comment by mbucchia** [comment]
Hi,

There's a whole bunch of people using PimaxXR + OpenXR Toolkit + DCS, and they're not having issues, so we need to find what's specific about your situation. 

- have you tried disabling OpenXR Toolkit?
- for the ETL stuff, the PimaxXR Control Center app should be prompting for elevation and run itself as administrator. However, I have heard of situation where a setting in Windows prevent that from happening.
- you can use the command line method: https://github.com/mbucchia/Pimax-OpenXR/wiki/Capturing-debug-traces
- Good catch on the `flags1` `flags2` mix up!
- I use 9-axis all the time without issues, so maybe not the right lead
- it does look like OpenComposite is passing the bad handle... what version of it are you using?

Thanks!

**Comment by daniellovell** [comment]
Ah, it was the OpenXR Toolkit version.. I swear I had read somewhere that only OpenXR Toolkit v1.1.0 was compatible with Pimax-OpenXR. 
I bumped to v1.1.4 and that fixed it, thanks!

**Comment by mbucchia** [comment]
Ha! Yeah the original version of PimaxXR did not like OpenXR Toolkit past 1.1.0, but this has been solved since!

Thanks for digging and finding the flags issue, I'll fix that one shortly!

**Comment by daniellovell** [comment]
No problem! There's still the strange corruption of the ETL output from WPR. If I stopped the trace before executing anything, the ETL file was fine. But if I stopped the trace via CLI or your companion app after the DCS crash, the ETL file would be corrupted. 
I imagine WPR should be robust to the traced code crashing. Has anyone else mentioned this?

**Comment by daniellovell** [comment]
Closed, resolved by updating OpenXR Toolking to v1.1.4

**Comment by mbucchia** [comment]
Flag typo issue fixed in edaf5e90722f1cad7996a8f97c1d6537dbb8c4f8.
